cmd/prom-migrator/main.go

```diff
@@ -135,8 +137,13 @@ func parseFlags(conf *config, args []string) {
 		"Setting this value less than zero will indicate all data from mint upto now. ")
 	flag.StringVar(&conf.maxBlockSize, "max-read-size", "500MB", "(units: B, KB, MB, GB, TB, PB) the maximum size of data that should be read at a single time. "+
 		"More the read size, faster will be the migration but higher will be the memory usage. Example: 250MB.")
-	flag.IntVar(&conf.numShards, "num-shards", 4, "Number of shards to split a fetched block in. Each shard is written concurrently. "+
+	flag.IntVar(&conf.concurrentPush, "num-shards", 4, "Number of shards to split a fetched block in. Each shard is written concurrently. "+
 		"Note: Larger shards count will lead to significant memory usage.")
+	flag.IntVar(&conf.concurrentPulls, "concurrent-pulls", 1, "Concurrent pulls enables fetching of data concurrently. "+
+		"This may enable higher throughput by pulling data faster from remote-read stores. "+
+		"Note: Setting concurrent-pulls > 1 will disable real-time transfer rate speed while fetching data. Moreover, setting this value too high may cause "+
```

Comment:
Not sure what this means.

Reply:
That is because I didn't update the code of this part earlier (as it was dependent on the previous PR). Now, it is updated and after testing it out, it looks pretty good (ui-ux) as explained here (explanation updated).